### PR TITLE
Setup: Introduced new file for database updates

### PR DESCRIPTION
**Pull request by mjansenDatabay**
Currently all database schema changes or database migrations have to be added to the `dbudate_04.php` file in `./setup/sql/`. The file currently contains **1184** (2018-10-26 13:30) single steps. I suggest to introduce a **new/empty** database file for future steps (**after** all contributors have commited their database update steps for 5.4.x on Monday).

`PhpStorm` likes this!

### CODE REVIEW

**Comment by klees** [comment]
In general I'm fine with this, but I'm planning to start with the Setup Revision (at least the thinking and discussions) soon, which imo will lead to getting rid of the current db_update-files anyway. Is there a specific reason to make this step just now?

**Comment by mjansenDatabay** [comment]
The reasons are:

1. My IDE needs too much resources ;-)
2. The 5.4.x release is a good time to switch the file
3. You even can't view the full file contents here: https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/setup/sql/dbupdate_04.php (only in raw mode or the latest *Firefox*)

**Comment by klees** [comment]
ad 3.: Try the new Firefox =)

File: Services/Database/classes/class.ilDBUpdate.php
Comment (legionth) [inline Services/Database/classes/class.ilDBUpdate.php]: Should be `5366` at least for trunk. I think

**Comment by legionth** [comment]
@klees sounds interesting :+1: Is this a feature that will come in the near future? I don't think that the feature development block this PR in any form. We have currently 5366 Update steps in here, which results in nearly 25k LoC and it's continually growing.

I don't know about you folks, but my IDE is very slow when editing/searching in this file. I appreciate a new file, so I won't be completely upset every time I open this file.

**Comment by klees** [comment]
We fixed the [general direction](https://docu.ilias.de/goto_docu_wiki_wpage_4900_1357.html) in the beginning of 2018 when Alex stepped down as setup maintainer. The discussions about multi-client and other projects left no time to take further steps, but I really try to pick up the discussion this year to implement at least some parts of the revision for ILIAS 6.0.

vim opens the file in the blink of an eye, and grep searches it without hesitation. So no problems here =)

**Comment by gvollbach** [comment]
@klees 
The file is too big now, what will happen in the future is not the solution for this problem. 

I also don't really get why the new file should be a problem for another solution in the future and in my opinion the discussion about the setup revision does not really belong to this pull request, since as you said, the discussion and thinking about this step will start in the future.

**Comment by legionth** [comment]
I am not sure to force developers to use other tools then their preferred tools is the way to go. Especially when the project is an Open Source Project and can be used/edited by multiple platforms and tools.

Could you explain why this should be blocked now for an upcoming feature that is, correct me if I'm wrong, in an conceptual state?

**Comment by mbecker-databay** [comment]
I'd say it's a "non-breaking change": The old file will still be too large for a growing number of standard-tools. But then, the new dbupdate-file will be fast and shiny.

So in the sense "The amount of dirt we can deal with is only limited by the number of available carpets", I support this PR as a quality-of-life enhancement, knowing Richard will fix the whole thing to something sustainable soon(ish).

**Comment by klees** [comment]
@gvollbach There is no problem for another solution in the future, I just tried to understand if there is any urgency in this PR. As the maintainer of the setup it is of course my duty to orchestrate the development of the component, so I cannot see why I should not point to other developments that are planned. The discussion about the general revision will of course not be made in this PR, though.

**Comment by klees** [comment]
@legionth Who is forcing anyone to use any tool? You was the one who asked about problems when editing or viewing the file. I have none. Also I'm not blocking anything. I was just trying to understand the urgency.

**Comment by klees** [comment]
@mbecker-databay The thing is exactly that: "Richard will fix the whole thing".

We may make this change for the new trunk, but I'm not convinced to make this change now, while people are commiting additions to the very same file that soon will be outdated. The setup currently is super-brittle anyway, so I'm afraid that I will have to fix (or at least look at) bugs that arise from whatever change in the setup or anywhere else anyway.

This is why I am a little reluctant about this.

**Comment by klees** [comment]
We talked about this in this in the @ILIAS-eLearning/technical-board and we are willing to merge this now since the trunk is open again. But I just realized that merging this also means that fixes in release_5-4 that contain DB-steps won't simply be mergeable to trunk. Should I still merge?

**Comment by bheyser** [comment]
For my feeling this step is the usual step we did in a certain interval. A new setup concept with another strategy to perform db updates would be nice, but for now this is indeed the right way for supporting our all dev tool phpstorm.

THX for the suggestion and the PR

**Review by bheyser — APPROVED** [review]

**Comment by bheyser** [comment]
@mjansenDatabay @klees I guess we need a 5_4_hotfix script since any dbupdate in 5.4.x is indeed a hotfix, because further alpha development must not take place in the 5.4.x branch.

IF we do NOT introduce this 5_4_hotfix script NO trunk development can be started for 6.0 when DBupdates are neccessary.

For my understanding a stable (or future stable) branch MUST NOT have the same dupdate script (!)

**Comment by klees** [comment]
@bheyser This sounds plausible. Thx!

@mjansenDatabay Agree?

**Comment by mjansenDatabay** [comment]
@klees Indeed, with the new release_5-4 branch we definitely need a hotfix script for 5.4.x related database schema changes or migrations.

**Comment by Amstutz** [comment]
TB: @mjansenDatabay, thanks a lot. Please adapt the step-index and merge.

**Pull request merged**
